### ISSUE TITLE
Losing your published URL when deleting the published version is unexpected.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#2291]
Just had a conversation with @taisdesouzalessa about her recent Thimble project. She had a bit of trouble sharing her link because she didn't realize that deleting the published version would cause Thimble to give her a new URL when she published again.

I think her expectation (and I agree) is that you'd keep your original published URL for the duration of your work on that project. Also, after deleting and re-publishing, the URL looks almost exactly the same except for one digit in the ID, so it's pretty hard to notice that it is different than your last one.

* Any reason that we can't "reserve" the published URL ID for each project after they are first published? 
* What if after publishing, the project stays live and instead of Deleting it, we just toggle some kind of redirect from that URL to a "project is no longer live" page?

cc @gideonthomas @humphd

**Comment by gideonthomas** [comment]
Adding the redirect is a bit hard. We've had this reported before and it required changing the way we deleted projects. I think once we switch to serverless storage, this shouldn't be a problem anymore

**Comment by flukeout** [comment]
Okay, maybe we should start some kind of feature / change doc so that we can make improvements when we switch over to the new system instead of designing for parity with existing stuff.

Do you have anything like that on the go - happy to start it if not. Maybe a doc would be good?

**Comment by taisdesouzalessa** [comment]
Yep, to add to Luke's comments my main problem was sharing it through Twitter. Once the new URL was not valid anymore I had to delete the tweet and create a new one with the updated URL. 

Otherwise it was a great and smooth experience. I really liked I could do changes in the code and see the updates reflected on the left panel without having to hit the "update" button.

**Comment by gideonthomas** [comment]
Glad to hear it @taisdesouzalessa!

@flukeout I don't have one specifically for that but I think we should start a bug vs. a doc with a list of parity needs and improvements.
